Core/clim-basic/utils.lisp

```diff
@@ -466,6 +466,17 @@ in KEYWORDS removed."
       (declare (dynamic-extent #',cont))
       (,fun ,@to-bind #',cont ,@to-pass))))
 
+;;; Macro writing
+
+(defmacro with-current-source-form ((form &rest more-forms) &body body)
+  "Associate errors signaled while executing BODY with FORM.
+MORE-FORMS are used when FORM is unsuitable (for example some
+implementations cannot associate errors with atoms). MORE-FORMS should
+be forms containing FORM."
+  #-sbcl (declare (ignore form more-forms))
+  #+sbcl `(sb-ext:with-current-source-form (,form ,@more-forms) ,@body)
+  #-sbcl `(progn ,@body))
+
```

Comment:
I'm not thrilled with an utility tailored specifically for a single implementation (with a stub for everything else). I gather it simplifies working with macros and I'll leave it be, but for runtime functionality we should discuss a policy of inclusion of implementation-specific magic. This remark may be considered as "resolved".

Reply:
> I'm not thrilled with an utility tailored specifically for a single implementation (with a stub for everything else). I gather it simplifies working with macros and I'll leave it be

The macro simplifies working with macros for the application programmer, not the macro writer. Specifically, it allows debugging invalid uses of macros more easily by allowing SLIME or other tools to highlight the sub-form of the whole macro form that is the cause of the problem. I will rewrite the commit message to state this more clearly.

In my opinion, more implementations should provide a similar functionality since it is very useful in programming interfaces with complex macros (other examples include `defclass` and `defpackage`). Therefore I hope it doesn't remain as implementation-specific. That said, I would like to try the feature in practice some more before writing a CDR. One piece of feedback I already got from your comment is that its purpose is not obvious without further explanation.

> but for runtime functionality we should discuss a policy of inclusion of implementation-specific magic.

I don't see how the distinction between macroexpansion and runtime is significant in this regard. Also, there is no magic involved. If the purpose of the change is unclear I simply have to explain it better.

> This remark may be considered as "resolved".

Sorry to put this bluntly, but then don't write it. I'm happy to answer questions regarding the change or acknowledge or address pointed out problems, but expressing dissatisfaction followed by "but it doesn't matter" seems inappropriate for a code review.

Reply:
macroexpansion time and runtime are different in a sense that I would ask you to remove the code if it had introduced a feature for applications  which works only on a single implementation (i.e if you had introduced automatic frame instance update which were working only on clisp).

Point taken, next time I'll either hit "cancel" button or ask for a change.